### ISSUE TITLE
Authorization headers (token based auth) & Socket interceptor

### ISSUE THREAD

**Issue by billyshena** [balderdashy/sails.io.js#56]
Hi everyone ! Currently using sails.io.js in my ReactJS app, I would like to know: 

1) If there is a way to add 'Auhtorization' headers (Bearer token) for my auth system (based on JWT).
2) How can I catch 401, 403 errors returned by my SailsJS API ? like the a http interceptor in Angular.js.

Thanks in advance !

Best,

**Comment by safani** [comment]
Currently to remedy this I send the headers as parameters and use pretty much the same logic server side to validate the socket with those credentials.  Sockets don't take headers.

**Comment by sailsbot** [comment]
Thanks for posting, @billyshena.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
